prow/test/integration/test/webhook_server_test.go

```diff
@@ -118,8 +125,142 @@ func TestWebhookServerMutateProwjob(t *testing.T) {
+		absDeploymentPath, err := filepath.Abs(deploymentPath)
+		if err != nil {
+			logrus.WithError(err).Info("could not find absolute path")
+		}
+		out, err := exec.Command("kubectl", "--context="+getClusterContext(), "apply", "-f", absDeploymentPath).CombinedOutput()
+		if err != nil {
+			logrus.WithError(err).Infof("could not clean up cluster %s", string(out))
+		}
+		out, err = exec.Command("kubectl", "--context="+getClusterContext(), "delete", "prowjob", testServerAvailability).CombinedOutput()
+		if err != nil {
+			logrus.WithError(err).Infof("prowjob not found to delete %s", string(out))
+		}
+		out, err = exec.Command("kubectl", "--context="+getClusterContext(), "delete", "prowjob", validBareProwJobID).CombinedOutput()
+		if err != nil {
+			logrus.WithError(err).Infof("prowjob not found to delete %s", string(out))
+		}
+	})
+
+	key := types.NamespacedName{
+		Namespace: "default",
+		Name:      "webhook-server",
+	}
+	var webhookDeployment appsv1.Deployment
+	err = kubeClient.Get(ctx, key, &webhookDeployment)
+	if err != nil {
+		t.Fatalf("could not get deployment %v", err)
+	}
+	originalProwJob := &v1.ProwJob{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "ProwJob",
+			APIVersion: "prow.k8s.io/v1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Labels: map[string]string{
+				"foo":               "bar",
+				"default-me":        "enabled",
+				"admission-webhook": "enabled",
+			},
+			Name:      testServerAvailability,
+			Namespace: "default",
+		},
+		Spec: v1.ProwJobSpec{
+			Cluster: "foo",
+			Refs: &v1.Refs{
+				Repo: "bar",
+			},
+			Agent:  v1.TektonAgent,
+			Type:   v1.PresubmitJob,
+			Report: true,
+		},
+		Status: v1.ProwJobStatus{
+			State: v1.PendingState,
+		},
+	}
+	selector, err := labels.Parse("app = webhook-server")
+	if err != nil {
+		t.Fatal("could not parse labels")
+	}
+	webhookDeploymentCopy := webhookDeployment.DeepCopy()
+	webhookDeploymentCopy.Spec.Template.Spec.Containers[0].LivenessProbe.PeriodSeconds = 4
+	var podList corev1.PodList
+	if err := kubeClient.List(ctx, &podList, &client.ListOptions{
+		LabelSelector: selector,
+	}); err != nil {
+		t.Fatalf("could not list pods %v", err)
+	}
+	var wg sync.WaitGroup
+	wg.Add(1)
+	oldName := podList.Items[0].GetName()
+	count := 0
+	go func() {
+		defer wg.Done()
+		for {
+			exist, newPod, err := checkNewPodInCluster(oldName, kubeClient, ctx)
+			if err != nil {
+				return
+			}
+			if exist && newPod.Status.Phase == "Running" {
+				return
+			}
+			//update foo label of prowjob, a cpoy is made to ensure there is no resource version
+			originalProwJobCopy := originalProwJob.DeepCopy()
+			originalProwJobCopy.Labels["foo"] = strconv.Itoa(count)
+			if err := kubeClient.Create(ctx, originalProwJobCopy, &client.CreateOptions{}); err != nil {
+				t.Errorf("prowjob could not be created. Error: %v ", err)
+				return
+			}
+			out, err := exec.Command("kubectl", "--context="+getClusterContext(), "delete", "prowjob", testServerAvailability).CombinedOutput()
```

Comment:
this is done in `Cleanup`, no need to duplicate here

Reply:
you cant create another prowjob without first deleting the current one

Reply:
ack.